src/core/coap/coap_header.cpp

```diff
@@ -281,6 +312,56 @@ ThreadError Header::AppendContentFormatOption(MediaType aType)
     return AppendOption(coapOption);
 }
 
+ThreadError Header::AppendMaxAgeOption(uint32_t aMaxAge)
+{
+    Option coapOption;
+    uint8_t buf[4];
+
+    coapOption.mNumber = kCoapOptionMaxAge;
+
+    if (aMaxAge <= 0xff)
+    {
+        buf[0] = static_cast<uint8_t>(aMaxAge);
+        coapOption.mLength = 1;
+    }
+    else if (aMaxAge <= 0xffff)
+    {
+        buf[0] = static_cast<uint8_t>(aMaxAge >> 8);
+        buf[1] = static_cast<uint8_t>(aMaxAge);
+        coapOption.mLength = 2;
+    }
+    else if (aMaxAge <= 0xffffff)
+    {
+        buf[0] = static_cast<uint8_t>(aMaxAge >> 16);
+        buf[1] = static_cast<uint8_t>(aMaxAge >> 8);
+        buf[2] = static_cast<uint8_t>(aMaxAge);
+        coapOption.mLength = 3;
+    }
+    else
+    {
+        buf[0] = static_cast<uint8_t>(aMaxAge >> 24);
+        buf[1] = static_cast<uint8_t>(aMaxAge >> 16);
+        buf[2] = static_cast<uint8_t>(aMaxAge >> 8);
+        buf[3] = static_cast<uint8_t>(aMaxAge);
+        coapOption.mLength = 4;
+    }
+
+    coapOption.mValue = buf;
+
+    return AppendOption(coapOption);
+}
+
+ThreadError Header::AppendUriQueryOption(const char *aUriQuery)
+{
+    Option coapOption;
+
+    coapOption.mNumber = kCoapOptionUriQuery;
+    coapOption.mLength = static_cast<uint16_t>(strlen(aUriQuery));
+    coapOption.mValue  = reinterpret_cast<const uint8_t *>(aUriQuery);
```

Comment:
Do you want me to replace  reinterpret to static cast in AppendUriPathOptions? Or keep the focus only on the new code?